### ISSUE TITLE
[aws-lambda-python] Automatic bundling not working correctly for aws_cdk.aws_lambda_python.PythonFunction when using the Python 3.8 runtime

### ISSUE THREAD

**Issue by CarlosDomingues** [aws/aws-cdk#9418]
[From the docs](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-python-readme.html):

> If `requirements.txt` exists at the entry path, the construct will handle installing all required modules in a Lambda compatible Docker container according to the `runtime`.

When not setting the `runtime` attribute or using `aws_lambda.Runtime.PYTHON_3_7`, the bundling works as expected. However, using `aws_lambda.Runtime.PYTHON_3_8` fails.

### Reproduction Steps

Create an aws_cdk.aws_lambda_python.PythonFunction object that uses the Python 3.8 runtime and needs an external dependency defined in `requirements.txt`:

`<projetct_root>/app.py`:

```python
from aws_cdk import (
    core,
    aws_lambda,
    aws_lambda_python
)

from os import path

class AwsCdkStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        my_lambda = aws_lambda_python.PythonFunction(
            self,
            "mylambda",
            entry=path.join(".", "functions", "mylambda"),
            index="lambda-handler.py",
            handler="handler",
            runtime=aws_lambda.Runtime.PYTHON_3_8,
        )

default_env = core.Environment(account="<account>", region="<region>")

app = core.App()
AwsCdkStack(app, "my-app", env=default_env)
app.synth()
```

`<projetct_root>/functions/mylambda/lambda-handler.py`:

```python
import requests as r

def handler(event, context):
     print(r.get("https://api.ipify.org?format=json").json())
```

`<projetct_root>/functions/mylambda/requirements.txt`:

```
requests==2.23.0
```
### Error Log

During the bundling I can see several warnings such as:

> WARNING: The directory '/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.

Then finally it fails with the following error:

> bash: rsync: command not found
> ....
> jsii.errors.JSIIError: Failed to run bundling Docker image for asset my-app/mylambda/Code/Stage: Error: docker exited with status 127
> Subprocess exited with error 1

### Environment

  - **CLI Version: 1.56.0 (build c1c174d)**
  - **Framework Version: 1.56.0**
  - **Node.js Version: v13.11.0** 
  - **OS: Windows 10**
  - **Language (Version): Python (3.8.2)**

**Comment by CarlosDomingues** [comment]
I made a quick test and it looks like a Docker image issue:

For `amazon/aws-sam-cli-build-image-python3.7:latest`:

```bash
$ docker run -it amazon/aws-sam-cli-build-image-python3.7:latest /bin/bash

bash-4.2# rsync --version
rsync  version 3.0.6  protocol version 30
Copyright (C) 1996-2009 by Andrew Tridgell, Wayne Davison, and others.
Web site: http://rsync.samba.org/
Capabilities:
    64-bit files, 64-bit inums, 64-bit timestamps, 64-bit long ints,
    socketpairs, hardlinks, symlinks, IPv6, batchfiles, inplace,
    append, ACLs, xattrs, iconv, symtimes

rsync comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
are welcome to redistribute it under certain conditions.  See the GNU
General Public Licence for details.
```

For `amazon/aws-sam-cli-build-image-python3.8:latest`:

```bash
$ docker run -it amazon/aws-sam-cli-build-image-python3.8:latest /bin/bash
bash-4.2# rsync --version
bash: rsync: command not found
```

**Comment by adamelmore** [comment]
Duplicate of #9349, fixed by #9355.

**Comment by eladb** [comment]
Closing as dup
